go/vt/mysqlctl/replication.go

```diff
@@ -174,8 +175,21 @@ func (mysqld *Mysqld) RestartReplication(hookExtraEnv map[string]string) error {
 }
 
 // GetMysqlPort returns mysql port
-func (mysqld *Mysqld) GetMysqlPort() (int32, error) {
-	qr, err := mysqld.FetchSuperQuery(context.TODO(), "SHOW VARIABLES LIKE 'port'")
+func (mysqld *Mysqld) GetMysqlPort(ctx context.Context) (int32, error) {
+	// We can not use the connection pool here. This check runs very early
+	// during MySQL startup when we still might be loading things like grants.
+	// This means we need to use an isolated connection to avoid poisoning the
+	// DBA connection pool for further queries.
+	params, err := mysqld.dbcfgs.DbaConnector().MysqlParams()
+	if err != nil {
+		return 0, err
+	}
+	conn, err := mysql.Connect(ctx, params)
+	if err != nil {
+		return 0, err
+	}
+	defer conn.Close()
+	qr, err := conn.ExecuteFetch("SHOW VARIABLES LIKE 'port'", 1, false)
```

Comment:
I know this is not super important here, but should the context be passed down to the actual query? We've seen various cases where MySQL ends up being "stuck" for whatever reason not replying to incoming queries, and this would cause the `GetMysqlPort` function to hang indefinitely (instead of returning an error once `ctx` expires).

Reply:
@arthurschreiber Do you mean as a general feature / refactor? It's not possible to pass in a context at the moment in the MySQL connection handling that Vitess does.

That might be useful as a separate feature / change, but I think that's independent of what we're doing here?